File: python/383_Ransom_Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        magazineChars = {}
        for i in range (0, len(magazine)):
            if magazineChars.get(magazine[i]) != None:
                magazineChars[magazine[i]] += 1
            else:
                magazineChars[magazine[i]] = 1
        print(magazineChars)
        for i in range (0, len(ransomNote)):
            if (magazineChars.get(ransomNote[i]) != None
                and magazineChars.get(ransomNote[i])) > 0:
                magazineChars[ransomNote[i]] -= 1
            else:
                return False
        return True

File: cpp/383_Ransom_Note .cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> magazineChars;
        for (int i = 0; i < magazine.size(); ++i) {
            if (magazineChars.find(magazine[i]) != magazineChars.end()) {
                magazineChars[magazine[i]] += 1;
            } else {
                magazineChars[magazine[i]] = 1;
            }
        }
        for (int i = 0; i < ransomNote.size(); ++i) {
            cout << magazineChars[ransomNote[i]];
            if (magazineChars.find(ransomNote[i]) != magazineChars.end()
                 && magazineChars[ransomNote[i]] > 0) {
                magazineChars[ransomNote[i]] -= 1;
            } else {
                return false;
            }
        }
        return true;
    }
};